lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -51,6 +52,11 @@
             }
         }
 
+        // Splitting IndexOf in several smaller methods make it passing the requirements for inlining.
```

Comment:
I agree. That is what I suggested to @LouisLaf . In long term it will decrease the maintainability. I would suggest to bypass the inline required conditions for all JS Built In function or make an option in the FunctionsEnum to specify to force inlining a specific built-in method.

---
In reply to: [122268005](https://github.com/Microsoft/ChakraCore/pull/3141#discussion_r122268005) [](ancestors = 122268005)